### ISSUE TITLE
No warning if additional clocks aren't configured

### ISSUE THREAD

**Issue by MichaelBell** [TinyTapeout/tt-gds-action#15]
It's not uncommon for people to think that they can clock registers from generated clocks or different input pins, without adding the required configuration to the SDC file.

Currently, no obvious warnings are flagged if you do this, but it is likely your design won't work due to hold violations (setup violations might also be an issue, but would generally not be design breaking as you can just clock more slowly).  This is because STA doesn't understand the clocks you are using so doesn't know to check the paths between different registers in your addition clock domains.

I think this problem will be picked up by there being "unclocked" registers in the STA report.  So it might be a good idea to grep for
`Warning: There are \d* unclocked register/latch pins.`
in multi_corner_sta.checks.rpt and flag a warning on the action summary if this is present.

**Comment by urish** [comment]
Thanks! Any example of such project?

**Comment by MichaelBell** [comment]
An example is https://github.com/KennethWilke/tt06-cdc-fifo/actions/runs/8564552127

**Comment by KennethWilke** [comment]
I'm not sure if this is a concern for my project. If it is, then I'll have a next topic of study! :smile:

The `read_clock` and `write_clock` inputs to my design seem pretty similar to something like an I2C SCL input with a maximum speed of "good luck, why don't you try to find out!". Would an I2C clock input have the same concern?

**Comment by urish** [comment]
The issue with hold violations is that slowing down the clock doesn't help (AFAIK, only varying the supply voltage or temperature can help after the fact). 

Check out this great video from @mattvenn, it does a good job of explaining hold violations: https://www.youtube.com/watch?v=5PRuPVIjEcs

**Comment by KennethWilke** [comment]
Will take a look at that video, and I'd gladly take any suggested improvements to my design. I'm hoping to revise it 1-2 times if I can improve test coverage and/or reliability.

I expect my FIFO should have similar challenges as https://github.com/cmu-stuco-98154/f22-tt02-jrecta/tree/main from TT02 as they aim to implement roughly the same thing

**Comment by urish** [comment]
I think it might make sense to add this to our "--print-warnings" that we already have in tt-support-tools:

https://github.com/TinyTapeout/tt-support-tools/blob/1ebe1ef29b33a27406877eaf3f15fd1b79f594ed/project.py#L695-L710

WDYT @MichaelBell ?

**Comment by MichaelBell** [comment]
Yes, that looks like the place to add the check.  I don't see any warnings in that multi_corner_sta.checks.rpt on a good run, so should probably just flag any warnings in that file, as is done for synthesis.

**Comment by urish** [comment]
So grep for "Warning: " ?
